EIPS/eip-draft-trie-penalty.md

```diff
@@ -0,0 +1,261 @@
+| EXTCODESIZE| Yes      | `extcodesize(inexistant_addr)` would incur `penalty`|
+| CALL | Yes| See details below about call variants|
+| CALLCODE | Yes| See details below about call variants|
+| DELEGATECALL | Yes| See details below about call variants|
+| STATICCALL | Yes| See details below about call variants|
+| SELFDESTRUCT | No| See details below. | 
+| CREATE   | No | Create destination not explicitly settable, and assumed to be inexistant already.|
+| CREATE2  | No | Create destination not explicitly settable, and assumed to be inexistant already.| 
+
+
+### Notes on Call-derivatives
+
+
+A `CALL` triggers a lookup of the `CALL` destination address. The base cost for `CALL` is at `700` gas. A few other characteristics determine the actual gas cost of a call:
+
+1. If the `CALL` (or `CALLCODE`) transfers value, an additional `9K` is added as cost. 
+    1.1 If the `CALL` destination did not previously exist, an additional `25K` gas is added to the cost.
+
+This EIP adds a second rule in the following way:
+
+2. If the call does _not_ transfer value and the callee does _not_ exist, then `penalty` gas is added to the cost.
+
+In the table below, 
+- `value` means non-zero value transfer, 
+- `!value` means zero value transfer, 
+- `dest` means destination already exists, or is a `precompile`
+- `!dest` means destination does not exist and is not a `precompile`
+
+| Op | value,dest| value, !dest |!value, dest| !value, !dest|  
+| -- | ---------  | -- | --| -- |
+|CALL |  no change | no change| no change| `penalty`|
+|CALLCODE |  no change | no change| no change| `penalty`|
+|DELEGATECALL | N/A | N/A| no change| `penalty` |
+|STATICCALL | N/A | N/A| no change| `penalty` |
+
+Whether the rules of this EIP is to be applied for regular ether-sends in `transactions` is TBD. See the 'Backwards Compatibility'-section for some more discussion on that topic. 
+
+### Note on `SELFDESTRUCT`
+
+The `SELFDESTRUCT` opcode also triggers an account trie lookup of the `beneficiary`. However, due to the following reasons, it has been omitted from having a `penalty` since it already costs `5K` gas. 
+
+### Clarifications:
+
+- The `base` costs of any opcodes are not modified by the EIP.
+- The opcode `SELFBALANCE` is not modified by this EIP, regardless of whether the `self` address exists or not. 
+
+### Determining the `penalty`
+
+A transaction with `10M` gas can today cause ~`14K` trie lookups.
+
+- A `penalty` of `1000`would lower the number to ~`5800` lookups, `41%` of the original.
+- A `penalty` of `2000`would lower the number to ~`3700` lookups, `26%` of the original.
+- A `penalty` of `3000`would lower the number to ~`2700` lookups, `20%` of the original. 
+- A `penalty` of `4000`would lower the number to ~`2100` lookups, `15%` of the original. 
+
+There exists a roofing function for the `penalty`. Since the `penalty` is deducted from `gas`, that means that a malicious contract can always invoke a malicious relay to perform the trie lookup. 
```

Comment:
Maybe it is worth considering the idea which was brought up as part of "Stateless Summit". Have a separate counter, this time for the entire transaction frame, for the penalty counting. This same counter could be used for witness size counting.

Wouldn't it eliminate this attack vector?